### ISSUE TITLE
Extend keptn delete service with --purge flag

### ISSUE THREAD

**Issue by johannes-b** [keptn/keptn#2776]
As a user, I can use the `keptn delete service` CLI command to delete an onboarded service. However, this command does not delete the Keptn CloudEvents related to this service. Consequently, I get the entire service history when onboarding a previously deleted service. 

**Improvement**

I propose to add the `--purge` flag to the `keptn delete service` command in order to remove the service-related events from the data store. 

**Tasks**
- Add the flag to the command: https://github.com/keptn/keptn/blob/master/cli/cmd/delete_service.go#L68
- Provide in-code documentation of this flag
- Extend the Keptn API endpoint (`/project/{projectName}/service/{serviceName}`) by an additional query parameter: `purge=true`
- Delete events from MongoDB
- Clean-up of materialized views

**Definition of Done**
- [ ] `keptn delete service xyz -p=abc --purge` removes every event related to the service. Consequently, another `keptn onboard service xyz` does not show the history of the previously deleted service xyz. 
- [ ] Closed #1795

**Comment by ankitjain28may** [comment]
I would like to work on this, can you please assign this to me.

**Comment by christian-kreuzberger-dtx** [comment]
Thanks! If you have any questions, don't hesitate to ask us on Slack (I recommend @warber @bacherfl as the primary contacts for this issue)

**Comment by christian-kreuzberger-dtx** [comment]
Hi @ankitjain28may  are you still looking into this issue? If not, we would like to re-assign it.

**Comment by ankitjain28may** [comment]
@christian-kreuzberger-dtx I am looking into this issue, can you please help me with the doc where I can find the routes for the API?

**Comment by christian-kreuzberger-dtx** [comment]
That's a bit hidden :D You can find it in shipyard controller

https://github.com/keptn/keptn/blob/master/shipyard-controller/handler/servicehandler.go

**Comment by ankitjain28may** [comment]
@christian-kreuzberger-dtx I am trying to reproduce the above issue but facing this issue while deleting the existing service
I have the project name `grafana` and service with name `grafana` with deployment strategy `direct`. Now I am trying to delete it via keptn CLI version 0.8 (latest master) but I create a service via keptn CLI v0.7.2

```
Delete project was unsuccessful
Error: Delete project was unsuccessful. could not delete service grafana from stage production: Stage  production does not exist.
```

**Comment by christian-kreuzberger-dtx** [comment]
Not sure what is going on there, but delete project should not be needed for this task anyway, right?

**Comment by ankitjain28may** [comment]
Not deleting the project, deleting the service just to see what happens. Let me try it with CLI v0.7.x

**Comment by christian-kreuzberger-dtx** [comment]
I've tried both with a recent build (see https://github.com/keptn/keptn/releases ) and delete service aswell as project seems to work fine.

**Comment by ankitjain28may** [comment]
Cool, let me try again then. Thanks @christian-kreuzberger-dtx

**Comment by gabe-villarreal** [comment]
This is a feature I'd like to see. is this slated for 0.8.1?

**Comment by warber** [comment]
Actually, I was expecting that the deletion of a service will actually indeed delete its related events. I would see this as a default behavior that could be overruled by a `--keep-events` flag or something like that. 
Are there specific reasons for keeping the events?

**Comment by johannes-b** [comment]
Hi @ankitjain28may, 
how is it going with this issue? Are you still working on it?
